src/MCPClient/lib/clientScripts/parse_dataverse_mets.py

```
@@ -0,0 +1,265 @@
+    if checksum != generated_checksum:
+        job.pyprint('Checksum failed')
+        event_outcome = "Fail"
+        detail_note = 'Dataverse checksum %s verification failed' % checksum
+    else:
+        job.pyprint('Checksum passed')
+        event_outcome = "Pass"
+        detail_note = 'Dataverse checksum %s verified' % checksum
+
+    databaseFunctions.insertIntoEvents(
+        fileUUID=file_uuid,
+        eventIdentifierUUID=event_id,
+        eventType='fixity check',
+        eventDateTime=date,
+        eventDetail=event_detail,
+        eventOutcome=event_outcome,
+        eventOutcomeDetailNote=detail_note,
+    )
+
+
+def parse_(job, unit_path, unit_uuid):
+    """Access the existing METS file and extract and validate its components.
+    """
+    dataverse_mets_path = os.path.join(unit_path, 'metadata', 'METS.xml')
+    mets = metsrw.METSDocument.fromfile(dataverse_mets_path)
+    mapping = get_db_objects(job, mets, unit_uuid)
+    if mapping is None:
+        logger.error(
+            "Exiting. Returning the database objects for our Dataverse "
+            "files has failed.")
+        return 1
+    update_file_use(job, mapping)
+    agent = add_external_agents(job, unit_path)
+    create_derivatives(job, mapping, agent)
+    validate_checksums(job, mapping, unit_path)
+    return 0
+
+
+def parse_dataverse_mets(job):
+    """Extract the arguments provided to the script and call the primary
+    function concerned with parsing Dataverse METS.
+    """
+    try:
+        transfer_dir = job.args[1]
+        transfer_uuid = job.args[2]
+        logger.info("Parse Dataverse METS with dir args: '%s' and transfer "
+                    "uuid: %s", transfer_dir, transfer_uuid)
+        return parse_(job, transfer_dir, transfer_uuid)
+    except IndexError:
+        logger.error(
+            "Problem with the supplied arguments to the function "
+            "len: %s", len(job.args))
+        return 1
```

We have a lot of functions in client script doing this same thing, returning exit codes. I'm thinking it'd be much nicer if we separate concerns, e.g. raise an exception instead of returning 1 (`JobContext` will automatically capture and set the status to 1 for us) or just don't return or return `None` which will result in 0.

---

I think your suggestion is neater so will try that out in these scripts. It'll be good to find the consistency in the other scripts over time.

---

When you take a look at the modified PR, I'm not sure if I've addressed all your concerns though, but happy to make changes accordingly.